sources/api/bootstrap-commands/src/main.rs

```diff
@@ -0,0 +1,315 @@
+    let config: BootstrapCommandConfig =
+        toml::from_str(&config_str).context(error::DeserializationSnafu {
+            config_path: config_path.as_ref(),
+        })?;
+
+    Ok(config.bootstrap_commands.unwrap_or_default())
+}
+
+/// Parse the args to the program and return an Args struct
+fn parse_args(args: env::Args) -> Result<Args> {
+    let mut global_args = Args::default();
+
+    let mut iter = args.skip(1);
+    while let Some(arg) = iter.next() {
+        match arg.as_ref() {
+            // Global args
+            "--log-level" => {
+                let log_level = iter.next().context(error::UsageSnafu {
+                    message: "Did not give argument to --log-level",
+                })?;
+                global_args.log_level = LevelFilter::from_str(&log_level)
+                    .context(error::LogLevelSnafu { log_level })?;
+            }
+
+            "-c" | "--config-path" => {
+                let config_str = iter.next().context(error::UsageSnafu {
+                    message: "Did not give argument to --config-path",
+                })?;
+                global_args.config_path = PathBuf::from(config_str.as_str());
+            }
+
+            _ => (),
+        }
+    }
+
+    Ok(global_args)
+}
+
+fn run() -> Result<()> {
+    let args = parse_args(env::args())?;
+
+    // SimpleLogger will send errors to stderr and anything less to stdout.
+    SimpleLogger::init(args.log_level, LogConfig::default()).context(error::LoggerSnafu)?;
+
+    let bootstrap_commands = get_bootstrap_commands(args.config_path)?;
+
+    // We want to run bootstrap commands in sorted order of their names.
+    for (name, bootstrap_command) in bootstrap_commands.iter() {
+        let essential = bootstrap_command.essential.unwrap_or(false);
+        let status = handle_bootstrap_command(name, bootstrap_command);
+
+        ensure!(
+            !essential || status.is_ok(),
+            error::BootstrapCommandFailureSnafu {
+                name: name.as_ref()
+            }
+        )
+    }
+
+    Ok(())
+}
+
+fn main() {
+    if let Err(e) = run() {
+        eprintln!("{}", e);
+        process::exit(1);
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+mod error {
+    use snafu::Snafu;
+    use std::path::PathBuf;
+    use std::process::Command;
+
+    #[derive(Debug, Snafu)]
+    #[snafu(visibility(pub(super)))]
+    pub(super) enum Error {
+        #[snafu(display("Failed to read settings from config at {}: {}", config_path.display(), source))]
+        ReadConfig {
+            config_path: PathBuf,
+            source: std::io::Error,
+        },
+
+        #[snafu(display("Failed to deserialize settings from config at {}: {}", config_path.display(), source))]
+        Deserialization {
+            config_path: PathBuf,
+            source: toml::de::Error,
+        },
+
+        #[snafu(display("Essential Bootstrap command with name: {} failed.", name))]
```

Comment:
nit: avoid saying "name" in error messages, and just give the quoted identifier:
```suggestion
        #[snafu(display("Essential bootstrap command '{}' failed.", name))]
```